### PR TITLE
lib/ini: better output

### PR DESCRIPTION
**Pull request by Morriar**
This PR changes three things:
* remove `nullable` warnings
* keep the keys in the same order than they are added to the config file
* group keys in the ini output

### CODE REVIEW

File: lib/ini/ini.nit [hunk 1]
```diff
@@ -235,7 +237,7 @@ class ConfigTree
 		key = key.substring(0, key.length - 2)
 		var len = 0
 		if has_key(key) then
-			len = get_node(key).children.length
+			len = get_node(key).as(not null).children.length
```
Comment (privat) [inline lib/ini/ini.nit]: `self[key]` seems to already return a nullable String. You may use it instead of `as(not null)`

File: lib/ini/ini.nit [hunk 2]
```diff
@@ -66,7 +66,7 @@ class ConfigTree
 	fun at(key: String): nullable Map[String, String] do
 		var node = get_node(key)
 		if node == null then return null
-		var map = new HashMap[String, String]
+		var map = new ArrayMap[String, String]
```
Comment (privat) [inline lib/ini/ini.nit]: `HashMap` already keeps the order if I remember correctly. Are you sure the order is not preserved?

File: lib/ini/ini.nit (hunk 1 above)
Comment (Morriar) [reply lib/ini/ini.nit]: I need a ConfigNode and not a String?

Comment (privat) [reply lib/ini/ini.nit]: then use `get_node` if you need a node :p

~~~nit
var node = get_node(key)
if node != null then len = node.children.length
~~~

**Comment by Morriar** [comment]
reroll @privat: removed useless cast

**Pull request merged**